src/ripple/app/tests/Env.h

```diff
@@ -0,0 +1,249 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2012-2015 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_TESTS_ENV_H_INCLUDED
+#define RIPPLE_APP_TESTS_ENV_H_INCLUDED
+
+#include <ripple/app/tests/Common.h>
+#include <ripple/app/tests/JTx.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/json/json_value.h>
+#include <ripple/json/to_string.h>
+#include <ripple/protocol/Issue.h>
+#include <ripple/protocol/RippleAddress.h>
+#include <ripple/protocol/STAmount.h>
+#include <ripple/protocol/STObject.h>
+#include <ripple/protocol/STTx.h>
+#include <beast/unit_test/suite.h>
+#include <boost/logic/tribool.hpp>
+#include <utility>
+#include <unordered_map>
+
+namespace ripple {
+namespace test {
+
+/** A view to an account's account root. */
+class AccountInfo
+{
+private:
+    Account account_;
+    std::shared_ptr<Ledger> ledger_;
+    std::shared_ptr<STLedgerEntry> root_;
+
+public:
+    AccountInfo(Account const& account,
+            std::shared_ptr<Ledger> const& ledger)
+        : account_(account)
+        , ledger_(ledger)
+        , root_(ledger->getAccountRoot(
+            account.id()))
+    {
+    }
+
+    STAmount
+    balance (Issue const& issue) const;
+
+    std::uint32_t
+    seq() const;
+
+    std::uint32_t
+    flags() const;
+};
+
+//------------------------------------------------------------------------------
+
+/** A transaction testing environment. */
+class Env
+{
+public:
+    beast::unit_test::suite& test;
+
+    /** The master account. */
+    Account const master;
+
+    /** The open ledger. */
+    std::shared_ptr<Ledger> ledger;
+
+public:
+    Env (beast::unit_test::suite& test_);
```

Comment:
Our naming convention is trailing underscores for non-public members and names of formal arguments used to initialize public members:
https://github.com/vinniefalco/rippled/blob/style/doc/CodingStyle.md#c